### PR TITLE
Single sites: Add a feature color behind the screenshots

### LINKED ISSUE #138: Single site featured images & color
_Split from #132._

Single sites have two featured images (this was added in #126), they should be displayed in the header of the page. Optionally there could be an accent color chosen.

| Example 1 | Example 2 | Example 3 |
|---|---|---|
| Hero-1 | Hero-2 | Hero |

@marko-srb What color should be used if there is no accent color set? Did we still want to auto-generate colors? (that required manual screenshots, so I still need a fallback color)

### PR DESCRIPTION
**Pull request by ryelle**
Fixes #138. This adds a field in the editor to set a custom color for each site. It can use the generated color from local images (if an image has been uploaded). If no image is found and no color is set, it falls back to `#f6f6f6`. This PR also updates the image sizes of the upload and the mshots request to use the latest viewport & image sizes.

Screenshot 2023-08-03 at 1 04 18 PM

cc @wordpress/meta-design — Does this editor flow make sense? Is this what you expected for color management? You can view the frontend screenshots too, but the spacing/etc is probably not right because it seems to be still in flux.

**Screenshots**

On the frontend, the color is used as the header background color.

| Default | Generated from local image | Manually set color |
|---|---|---|
| site-default | site-image | site-manual |

In the editor, the input can be found in the "Post" sidebar, under the Screenshots panel. If there is a local image, the toggle option to use it appears. If the generated color from the image looks bad, the toggle can be turned off and a custom color can be set. If there is no local image, the toggle does not appear, and the only option is custom or `#f6f6f6`.

| | Screenshot |
|---|---|
| No local image, default value | site-editor-default |
| No local image, custom value | site-editor-manual |
| Local image, uses image color | site-editor-image |
| Local image, override with custom color | site-editor-image-manual |

**To test**

You can test this all in one post, or create a few with different settings. This flow should hit all the major UI points.

1. Edit a site post
2. Set a color
3. View the post, it should use that color
4. Edit again
5. Reset the color
6. View the post, it should use `#f6f6f6`
7. Upload an image to use as the Desktop screenshot
8. The color section should display the active toggle
9. View the post, it should use a color from your image
10. Edit again, disable the toggle
11. View the post, it should use `#f6f6f6`
12. Edit again, set a custom color
13. View the post, it should use your custom color

Try various combinations of images, colors, etc.

### CODE REVIEW

**Review by StevenDufresne — APPROVED** [review]
Nice! I like the interface and this is working well.

**Comment by jasmussen** [comment]
Nice, thanks for trying this.

In the small nitpick category, the stroke around the devices looks a bit on the thin side, I guess that's separate, but CC: @fcoveram for awareness. The 10% black colored strokes also look a bit light here, and by the way we may want to make them a [stacked box-shadows](https://codepen.io/joen/pen/LYXMBZR) so the devices can potentially overlap on mobile. Francisco will also know whether we ended up doing that or not.

I'm not entirely sure if this image is enough for us to decide whether the auto-generated color slug is good enough:



Specifially the color looks dark and muted, and too saturated compared to the more neutral gray colors I would expect from a stone company. 

On the flipside, this interface makes it look so trivially easy to add a color, that it doesn't seem a big lift to add these manually. And as we also discussed separately, falling back to a light gray is in many cases absolutely fine, and depending on how good we can get the auto-color, likely preferred. Can you share a couple of more screenshots of the auto-color to help us decide?

Thanks for great work!

**Comment by fcoveram** [comment]
Replying to the initial question

> Does this editor flow make sense? Is this what you expected for color management? 

Yes, and yes.

To @jasmussen's point. I think it's slighlty thinner than mockups, but since spacings are ongoing work, I will wait for that review. The mobile view of this page has the final version updated where, yes, mobile device overlaps the desktop one.

**Comment by ryelle** [comment]
> In the small nitpick category, the stroke around the devices looks a bit on the thin side, I guess that's separate

Yes, the frontend screenshots are not "final" and were just to show the process.

> Can you share a couple of more screenshots of the auto-color to help us decide?

How about I merge this PR and then we can try some screenshots on the staging site? If the generated color is bad, I can pull that code out.

> Replying to the initial question […] Yes, and yes.

Thanks!

> I think it's slighlty thinner than mockups, but since spacings are ongoing work, I will wait for that review. 

Yes, I'm waiting on final word on spacings before going too in-depth on that aspect of the pages.

**Comment by jasmussen** [comment]
Go for it!

**Comment by ryelle** [comment]
> Can you share a couple of more screenshots of the auto-color to help us decide?

I've merged the PR and deployed it to the staging site, and uploaded some of the screenshots from the google drive. So you can see this on:

- [Noma](https://wordpress.org/showcase-v2/noma/)
- [The Line Hotels](https://wordpress.org/showcase-v2/the-line-hotels/)
- [Caeserstone](https://wordpress.org/showcase-v2/caeserstone/)
- [Dita](https://wordpress.org/showcase-v2/dita/)
- [Tonal](https://wordpress.org/showcase-v2/tonal/)

If you want to make any changes to this, can you open a new issue? Thanks!

**Comment by marko-srb** [comment]
@ryelle 

For desktop screenshot, there were polished / cut screenshots in the same folder here:

https://drive.google.com/drive/folders/1HTZxL8c_L7RlilcWztc-XZfPA6EkvOve?usp=drive_link

Please use those, as they don't have the scroller on the right...

I'll take a look on other stuff shortly. 

Thank youu!

**Comment by ryelle** [comment]
The cropped image size is a few pixels off, but I think it's fine once uploaded, I don't see an issue on those pages now that I've updated them

**Pull request merged**
